From d2d2f065f8f881ed21bfd66cfefa6f3f6360df7d Mon Sep 17 00:00:00 2001
From: Kai Liu <kaliu@qti.qualcomm.com>
Date: Mon, 15 Jun 2015 13:47:31 +0800
Subject: [PATCH 2/7] cfg80211: Reset beacon interval when stop AP fails

When SSR happens at WLAN driver, the cfg80211 stop AP can fail at driver.
Make sure that the beacon interval is reset, even when this API fails.

Change-Id: I459f55ce5f4bc44c4d0e20170bd50a83c2d609b4
Signed-off-by: Sameer Thalappil <sameert@codeaurora.org>
---
 net/wireless/ap.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/wireless/ap.c b/net/wireless/ap.c
index bdad1f9..91d02ac 100644
--- a/net/wireless/ap.c
+++ b/net/wireless/ap.c
@@ -25,8 +25,8 @@ int __cfg80211_stop_ap(struct cfg80211_registered_device *rdev,
 		return -ENOENT;
 
 	err = rdev_stop_ap(rdev, dev);
+	wdev->beacon_interval = 0;
 	if (!err) {
-		wdev->beacon_interval = 0;
 		memset(&wdev->chandef, 0, sizeof(wdev->chandef));
 		wdev->ssid_len = 0;
 		rdev_set_qos_map(rdev, dev, NULL);
-- 
1.7.9.5

